src/CowAmmModule.sol

```diff
@@ -0,0 +1,299 @@
+     * @param safe The address of the Safe that created the AMM.
+     * @param token0 The address of the first token in the pair.
+     * @param token1 The address of the second token in the pair.
+     * @param orderHash The hash of the conditional order that created the AMM.
+     */
+    event CowAmmCreated(Safe indexed safe, address indexed token0, address indexed token1, bytes32 orderHash);
+
+    /**
+     * @notice Emitted when an active CoW AMM is closed.
+     * @param safe The address of the Safe that closed the AMM.
+     * @param orderHash The hash of the conditional order that created the AMM.
+     */
+    event CowAmmClosed(Safe indexed safe, bytes32 orderHash);
+
+    /**
+     * @param _settler The address of the GPv2Settlement contract.
+     * @param _extensibleFallbackHandler The address of the `ExtensibleFallbackHandler`.
+     * @param _composableCow The address of `ComposableCoW`.
+     * @param _handler The address of the ConstantProduct AMM implementation for creating new CoW AMMs.
+     */
+    constructor(
+        GPv2Settlement _settler,
+        ExtensibleFallbackHandler _extensibleFallbackHandler,
+        ComposableCoW _composableCow,
+        IConditionalOrder _handler
+    ) {
+        // GPv2 specifics to make sure we set the right things and they're immutable!
+        SETTLER = _settler;
+        COW_DOMAIN_SEPARATOR = SETTLER.domainSeparator();
+        VAULT_RELAYER = address(SETTLER.vaultRelayer());
+
+        // ComposableCoW specifics
+        EXTENSIBLE_FALLBACK_HANDLER = _extensibleFallbackHandler;
+        COMPOSABLE_COW = _composableCow;
+        HANDLER = _handler;
+    }
+
+    /**
+     * @notice Creates a new CoW AMM with the given parameters.
+     * @param token0 The address of the first token in the pair.
+     * @param token1 The address of the second token in the pair.
+     * @param minTradedToken0 The minimum amount of token0 before the AMM attempts auto-rebalance.
+     * @param priceOracle The address of the price oracle to use for the AMM.
+     * @param priceOracleData The data to pass to the price oracle.
+     * @param appData The app data to pass to the AMM.
+     */
+    function createAmm(
+        IERC20 token0,
+        IERC20 token1,
+        uint256 minTradedToken0,
+        address priceOracle,
+        bytes calldata priceOracleData,
+        bytes32 appData
+    ) external {
+        // Assume the caller is a Safe
+        Safe safe = Safe(payable(msg.sender));
+
+        if (activeOrders[safe] != EMPTY_AMM_HASH) {
+            // If there is an active order, the user must close it before creating a new one
+            // This is not fool-proof as the user could create another AMM outside of this module
+            revert ActiveAMM();
+        }
+
+        _createAmm(safe, token0, token1, minTradedToken0, priceOracle, priceOracleData, appData);
+    }
+
+    /**
+     * @notice Replaces the active CoW AMM with a new one.
```

Comment:
Let's add a disclaimer that this won't work if the order was created outside this module. I worry that users could first start the AMM without the module and then try to delete an order with this: it would be bad as they would just create a second order.

```suggestion
     * @notice Replaces the active CoW AMM that was created with this module with a new one.
```